driver-core/src/main/java/com/datastax/driver/core/ErrorTracker.java

```diff
@@ -0,0 +1,50 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core;
+
+/**
+ * Interface for objects that are interested in monitoring precisely errors returned by all hosts.
+ *
+ * An implementation of this interface can be registered against a Cluster
+ * object trough the {@link Cluster#register} method, after which the
+ * {@link #notifyError(Host, Statement, Exception)} method will be called after each query of the driver to a Cassandra
+ * host with the error that has been returned for this operation.
+ */
+public interface ErrorTracker {
+
+    /**
+     * Method that is called after each request returning an error code from Cassandra.
+     *
+     * @param host the host that has been queried.
+     * @param statement the statement that has been sent.
+     * @param exception the exception the query caused.
+     */
+    public void notifyError(Host host, Statement statement, Exception exception);
```

Comment:
I'd like to avoid having the `ErrorTracker` being an abstract as it constraints quite a lot possibilities of extension..
I understand there is a separate logic in the core code distinguishing _ErrorTracker_ and _LatencyTracker_ but I think it is not such a bad thing since the Latency trackers are called and updated in much more different places in the code where there isn't really a need for a ErrorTracker to be called. The error tracker would basically be called for every request and so most of the time it would have nothing to do since I guess we consider that most of the queries will succeed